src/App.tsx

```diff
@@ -1,14 +1,69 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
+function filterTodos(todos: Todo[], todosActivityFilter: string, todosQuery: string) {
+  let resultTodos = [...todos];
+  const query = todosQuery.toLowerCase();
+
+  switch (todosActivityFilter) {
+    case 'completed':
+      resultTodos = resultTodos.filter(todo => todo.completed
+      && todo.title.toLowerCase().includes(query));
+      break;
+    case 'active':
+      resultTodos = resultTodos.filter(todo => !todo.completed
+      && todo.title.toLowerCase().includes(query));
+      break;
+    default:
+      resultTodos = resultTodos.filter(todo => todo.title.toLowerCase().includes(query));
+      break;
+  }
+
+  return resultTodos;
+}
 
 export const App: React.FC = () => {
+  const [todoShown, setTodoShown] = useState<Todo>();
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todosActivityFilter, setTodosActivityFilter] = useState('All');
```

Comment:
Also apply the type for `todosActivityFilter` here, for this piece of state

```suggestion
  const [todosActivityFilter, setTodosActivityFilter] = useState<SomeTimeYouCreated>('All');
```